### ISSUE TITLE
TypeError: The superclass is not a constructor.

### ISSUE THREAD

**Issue by maka-io** [PaulLeCam/react-leaflet#909]
## Bug report

**Before** opening an issue, make sure to read the [contributing guide](https://github.com/PaulLeCam/react-leaflet/blob/master/CONTRIBUTING.md) and understand this is a bug tracker, not a support platform.

Please make sure to check the following boxes before submitting an issue.\
**Issues opened without using this template will be closed unless they have a good reason not to follow this template.**

- [ x] All peer dependencies are installed: React, ReactDOM and Leaflet.
- [ x] Using a supported version of React and ReactDOM (v17.0.0 minimum).
- [ x] Using the supported version of Leaflet (v1.7.1 minimum) and its corresponding CSS file is loaded.
- [ x] Using the [latest v3 version of React-Leaflet](https://github.com/PaulLeCam/react-leaflet/releases).
- [ x] The issue has not already been reported.
- [ x] Make sure you have followed the [quick start guide](http://leafletjs.com/examples/quick-start.html) for Leaflet.
- [ x] Make sure you have fully read the [documentation](https://react-leaflet.js.org/docs/start-introduction) and that you understand the [limitations](https://react-leaflet.js.org/docs/start-introduction#limitations).

### Expected behavior
Able to extend GridLayer.

```
import { GridLayer, withLeaflet } from "react-leaflet";

class MapboxGlLayerComponent extends GridLayer {
  createLeafletElement(props) {
    return L.mapboxGL(props);
  }
}
```

### Actual behavior
Errors, and GridLayer is not defined.

```
console.log(GridLayer);
undefined
```

TypeError: The superclass is not a constructor.

### Steps to reproduce
Basically, I took this code, and updated everything to the latest version and got the same error.
https://codesandbox.io/s/react-leaflet-mapbox-forked-z1zvl?file=/src/MapboxLayer.js

This does work in version 2.8.0.
